learning_assistant/views.py

```diff
@@ -380,8 +296,19 @@ def get(self, request, course_run_id):
         # return no messages in the response.
         message_history_data = []
 
+        course_data = get_cache_course_run_data(course_run_id, ['start', 'end'])
+        today = datetime.now()
+        start = course_data.get('start', None)
+        end = course_data.get('end', None)
+
+        valid_dates = (
+            (start <= today if start else True)
+            and (end >= today if end else True)
+        )
+
         has_trial_access = (
-            enrollment_mode in valid_trial_access_modes
+            valid_dates
+            and enrollment_mode in valid_trial_access_modes
```

Comment:
I can't comment on line 318, but could you remove the parentheses in `(enrollment_mode in valid_full_access_modes)`? They're not necessary; I'm not sure why I added them originally.